openlibrary/plugins/upstream/models.py

```diff
@@ -111,6 +113,16 @@ def get_ia_cover(self, itemid, size):
         w, h = image_sizes[size.upper()]
         return "https://archive.org/download/%s/page/cover_w%s_h%s.jpg" % (itemid, w, h)
 
+    def get_isbn13(self):
+        """Fetches either isbn10 or isbn13 from record and returns canonical
+        isbn13
+        """
+        isbn_13 = self.isbn_13 and canonical(self.isbn_13[0]) or ""
+        if not isbn_13:            
+            isbn_10 = self.isbn_10 and self.isbn_10[0] or ""
+            return isbn_10 and canonical(isbn_10_to_isbn_13(isbn_10))
```

Comment:
`isbn_10_to_isbn_13` should already be returning a canonical isbn here

Reply:
I don't see canonical being run in utils/isbn

Reply:
It looks like it runs it on the incoming isbn, so the result will always be canonicalized.

Reply:
Done